### ISSUE TITLE
Coordinate release of compiler + Bower components in the future to save newbie pain

### ISSUE THREAD

**Issue by Andrew-Webb** [purescript/purescript#1927]
I started with the first [learning page](http://www.purescript.org/learn/getting-started/) today, carefully following the instructions, and dealing with the various problems that arose relating to `pulp init` (Bower not installed; couldn't find Git, etc.)  No biggie.

Finally everything was ready... but then `pulp build` gave me 50 warnings...(!!) whereas the Getting Started text anticipated none.  This was very off-putting.

Spelunking around, I now realize that the compiler is currently in advance on the `console`, `eff` and `prelude` components, which are still old-skool.

Suggestion: in future, coordinate the release of the compiler + these 3 basic components to save this kind of pain for newbies like me.

**Comment by Andrew-Webb** [comment]
plus there were loads more warnings once I'd followed the instruction to install `purescript-lists`

**Comment by garyb** [comment]
It's a little easier said than done. There are something like 70 core libraries to update, and we're also trying to do a bunch of major updates to fix various things that need breaking changes so that we don't have to do it again for a while. 

The whole 0.8 release hasn't been ideal, really. Everything took longer than we'd intended. When it's mostly handled by two people (not to discount all the great contributions we get) it's hard to get everything done quickly and in sync.

**Comment by user471** [comment]
>It's a little easier said than done. There are something like 70 core libraries to update,

Would it be more practical to create some tool `psc-migrate` and every time when there is a new deprecation or breaking change add some rules to convert source files? I think creating such tool could take less time than manually fix 70 core libraries.

**Comment by AppShipIt** [comment]
I've identified strategies to protect the site and book from bit rot but libraries remain an outlier. Maybe a "core" team within the GitHub organization? It could be an opportunity to engage the community with high impact contributions through low hanging (or mechanical) changes.

**Comment by AppShipIt** [comment]
@user471 I'm under the impression @garyb does something similar now. I don't see bandwidth for something official till "1.0" though.

**Comment by kRITZCREEK** [comment]
Bumping the version number of a library like prelude is going to make every other purescript package have out-of-date dependencies.  As long as the purescript environment is still as volatile as it is right now we should put up with the warnings (which can be hidden with tooling) over creating a dependency disaster.

**Comment by AppShipIt** [comment]
@kRITZCREEK I can't remember where but someone also requested not propagating warnings from dependent packages. It seems that'd require the compiler to know about packages though. `psa` could based on path right?

**Comment by hdgarrood** [comment]
Yes, `psa` already does this with the `--censor-lib` flag.

**Comment by AppShipIt** [comment]
Thanks @hdgarrood I need to try it!

**Comment by garyb** [comment]
The changes being made to the libraries aren't purely fixing warnings, so we can't use a tool to do it automatically. @telser is doing an admiral job of opening PRs for that for almost everything everywhere, but we're trying to fix up outstanding open issues at the same time, etc.

**Comment by user471** [comment]
>The changes being made to the libraries aren't purely fixing warnings, so we can't use a tool to do it automatically.

Yes, but Andrew-Webb is talking about 50 warnings after `pulp init` and all of them an deprecated syntax warnings.

**Comment by telser** [comment]
> > The changes being made to the libraries aren't purely fixing warnings, so we can't use a tool to do it automatically.
> Yes, but Andrew-Webb is talking about 50 warnings after pulp init and all of them are deprecated syntax warnings.

@user471 There are quite a few warnings, and it is unfortunate hence the PRs I've opened to help with that. The purescript team is getting the warnings and other issues handled at once before bumping library versions though. To get rid of the warnings without using the dependency suppression mentioned before, not only does the dep version need to be updated but everything that depends on it. 

In short-term the warnings are present, with what I see as three options 

1. Use the psa flag that @hdgarrood mentioned above
2. Stick to a purescript-0.7.x release
3. Ignore the warnings

None of those address the new user issue @Andrew-Webb initially brought up, given the defaults in pulp and the book documentation. But that's what we have *right* now.

Disclaimer: I'm not a member of the core purescript team and can't speak to the timetables of when things will be fixed, just what I have seen and how I have attempted to make the situation smoother on everyone. I think purescript is great and the people behind it are doing a good job including not marking it as a 1.0 release yet.
